client/src/components/FileSources/Instances/EditInstance.vue

```diff
@@ -36,8 +37,17 @@ const loadingMessage = "Loading file source template and instance information";
 async function onSubmit(formData: any) {
     if (template.value) {
         const payload = editFormDataToPayload(template.value, formData);
-        const args = { user_file_source_id: String(instance?.value?.uuid) };
-        const { data: fileSource } = await update({ ...args, ...payload });
+        const user_file_source_id = instance?.value?.uuid!;
```

Comment:
Does it compile if you do `if (template.value && instance.value)` and then remove the non-null operator ?
It seems a little clearer that way that onSubmit is only called if both of these are defined (though a comment wouldn't hurt).

Reply:
Yep, that was an oversight. Thanks for the catch!